toolkits/slack/arcade_slack/tools/chat.py

```diff
@@ -490,67 +579,132 @@ async def get_conversation_metadata_by_id(
     return dict(**extract_conversation_metadata(response["channel"]))
 
 
-@tool(
-    requires_auth=Slack(
-        scopes=["channels:read", "groups:read", "im:read", "mpim:read"],
-    )
-)
-async def get_conversation_metadata_by_name(
+@tool(requires_auth=Slack(scopes=["channels:read"]))
+async def get_channel_metadata_by_name(
```

Comment:
Renaming from `get_conversation_metadata_by_name` to `get_channel_metadata_by_name` is in effect removing a tool (breaking change) without any deprecation warning. 

It would probably be better if we deprecated the tool with a warning & provide a transition period.

Reply:
true, we can actually keep both for some time and have the old tool call the new one.. it's just that the name was not correct, since it couldn't really retrieve metadata from any conversation, only from channels, so the name change was meant to better express that

Reply:
Just to document: we ended up concluding that the warnings need to be raised in the runtime where the tool execution is requested, not where it actually runs. Which means, we need a way for tools to pipe warning messages to our SDKs through the engine & worker, so that the SDKs can raise the warnings to the developers using them.

We'll move forward with this PR without warnings, but are starting to work on an implementation for this warning infrastructure.